gcloud/storage/_helpers.py

```diff
@@ -52,12 +52,20 @@ def __init__(self, name=None):
         self._properties = {}
         self._changes = set()
 
-    def reload(self):
-        """Reload properties from Cloud Storage."""
+    def reload(self, connection=None):
+        """Reload properties from Cloud Storage.
+
+        :type connection: :class:`gcloud.storage.connection.Connection`
+        :param connection: An explicit connection to use for the API request.
+                           If not passed, use the connection assigned to
+                           the object in its constructor.
+        """
+        if connection is None:
+            connection = self.connection
```

Comment:
I had thought we might make a subsequent pass, removing all the `self.connection` usage (and the existence of a connection attribute) throughout the three APIs.  I can do the change here, instead, if you think that best.